.github/workflows/docker-develop.yml

```diff
@@ -49,3 +50,34 @@ jobs:
         run: |
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+      - name: Install SSH Key
```

Comment:
Admittedly, I don't know much about ansible and co, but have the private ssh key accessible from GHA seems like a security risk as someone could create a PR that does anything on the target machine.

Reply:
As far as I know, we can add a review access policy for GitHub secrets:
https://docs.github.com/en/actions/reference/encrypted-secrets#reviewing-access-to-organization-level-secrets
It means that GHA cannot access the secret until you approve it.

Alternatively, we could create a user with the lowest privileges on host machines and allow GHA to connect via this user and not the root. This way the impact on host machines from GHA will be minimum 
@karimodm

Reply:
@Wollac one more thing, here is a note from GHA docs:

> Note: With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

So it means that only people who have **write** access to the Goshimmer repo can access secrets.

Reply:
> So it means that only people who have write access to the Goshimmer repo can access secrets.
But still, there are a lot more people with Repo access than people that should have access on the machine

Reply:
Then we should try GHA review access feature for secrets.

Reply:
Ideally would be to restrict edit access to GHA workflow files, but it seems not possible for now. Here is an open issue requesting that:
https://github.com/actions/runner/issues/458

Reply:
In any case; I don't think this observation should be a blocker for the approval of this PR. We will have a separate discussion.